initialize/auto/auto.go

```diff
@@ -0,0 +1,21 @@
+// Package auto does automatic detection and runtime configuration for certain
+// environments.
+package auto
+
+import (
+	"context"
+)
+
+var msgs = []func(context.Context){}
+
+func init() {
+	CPU()
+}
+
+// Logs uses zlog to report any messages queued up during initialization.
+func Logs(ctx context.Context) {
```

Comment:
Bit of a weird one as it's not returning anything or really changing the state but maybe a name that's more imperative. What to you think?